apps/src/templates/sectionAssessments/sectionAssessmentsRedux.js

```diff
@@ -384,6 +384,66 @@ export const getStudentsMCSummaryForCurrentAssessment = (state) => {
   return studentsSummaryArray;
 };
 
+// Returns an array of objects corresponding to each question and the
+// number of students who answered each answer.
+export const getMultipleChoiceSectionSummary = (state) => {
+  // Set up an initial structure based on the multiple choice assessment questions.
+  // Initialize numAnswered for each answer to 0, and totalAnswered for each
+  // question to 0.
+  const assessmentsStructure = getCurrentAssessmentQuestions(state);
+  if (!assessmentsStructure) {
+    return [];
+  }
+  const questionData = assessmentsStructure.questions;
+  const multiQuestions = questionData.filter(question => question.type === 'Multi');
+  // TODO(caleybrock): Follow up to calculate multipleChoiceOption consistently.
+  const results = multiQuestions.map(question => {
+    return {
+      id: question.level_id,
+      question: question.question_text,
+      answers: question.answers.map((answer, index) => {
+        return {
+          multipleChoiceOption: ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H'][index],
+          isCorrect: answer.correct,
+          numAnswered: 0,
+        };
+      }),
+      totalAnswered: 0,
+      notAnswered: 0,
+    };
+  });
+
+  // Calculate the number of students who answered each option and fill
+  // in the initialized results structure above.
+  const studentResponses = getAssessmentResponsesForCurrentScript(state);
+  if (!studentResponses) {
+    return [];
+  }
+  Object.keys(studentResponses).forEach(studentId => {
+    studentId = parseInt(studentId, 10);
+    const studentObject = studentResponses[studentId];
+    const currentAssessmentId = state.sectionAssessments.assessmentId;
+    const studentAssessment = studentObject.responses_by_assessment[currentAssessmentId] || {};
+
+    const studentResults = studentAssessment.level_results || [];
+    const multiResults = studentResults.filter(result => result.status !== 'free_response');
```

Comment:
Once this PR merges, we'll merge staging, and use the constants instead of comparing string values directly: https://github.com/code-dot-org/code-dot-org/pull/23216